neurons/validator.py

```diff
@@ -1,38 +1,39 @@
 # ruff: noqa: E402
 import asyncio
 import time
-from typing import Optional
-
-import numpy as np
-
 from prompting import settings
 
 settings.settings = settings.Settings(mode="validator")
 settings = settings.settings
 
-import huggingface_hub
 from loguru import logger
-
-from neurons.forward import handle_response, log_stream_results
-from prompting.base.dendrite import DendriteResponseEvent, StreamPromptingSynapse
 from prompting.base.validator import BaseValidatorNeuron
-from prompting.datasets.base import BaseDataset
-from prompting.llms.vllm_llm import vLLMPipeline
-from prompting.tasks.base_task import BaseTask
+from neurons.forward import log_stream_results, handle_response
+from prompting.base.dendrite import DendriteResponseEvent, StreamPromptingSynapse
 from prompting.tasks.task_registry import TaskRegistry
-from prompting.utils.logging import ErrorEvent, ValidatorEvent, log_event
-from prompting.utils.uids import get_random_uids
+from prompting.utils.logging import log_event
+from prompting.utils.logging import ValidatorLoggingEvent, ErrorLoggingEvent
+from prompting.rewards.scoring import task_scorer
+from prompting.miner_availability.miner_availability import availability_checking_loop, miner_availabilities
+from prompting.llms.model_manager import model_scheduler
+from prompting.utils.timer import Timer
+from prompting.mutable_globals import scoring_queue
+from prompting import mutable_globals
 
-try:
-    from organic_scoring.synth_dataset import SynthDatasetConversation
+NEURON_SAMPLE_SIZE = 100
+SCORING_QUEUE_LENGTH_THRESHOLD = 10
 
-    from prompting.organic.organic_scoring_prompting import OrganicScoringPrompting
-except ImportError:
-    raise ImportError(
-        "Could not import organic-scoring library.  Please install via poetry: `poetry install --extras 'validator'`"
-    )
+# will start rotating the different LLMs in/out of memory
+asyncio.run(model_scheduler.start())
 
-NEURON_SAMPLE_SIZE = 100
+# will start checking the availability of miners at regular intervals
+asyncio.run(availability_checking_loop.start())
+
+# start scoring tasks in separate loop
+asyncio.run(task_scorer.start())
```

Comment:
It will start 3 loops everytime validator is imported (even in notebooks), try to use main entry point
Is there a possibility to make inference more modular, so that is can be plugged in/out with 1 line?

Reply:
You can just remove it from the task_registry, which is kind of the "config" for which tasks we are using. Point about the loops is fair though!